JavaScript/Code quality.md

```diff
@@ -0,0 +1,215 @@
+      "prettier --write"
+    ]
+    "**/*.scss": [
+      "stylelint --syntax=scss"
+    ]
+  }
+}
+```
+
+What `lint-staged` does is it matches files to `glob` patterns and passes the list of files as an argument to scripts. Tooling developers should ensure that their scripts can receive the list of files in the correct format. Most common tools like eslint, tslint, stylelint and others are compatible with the way `lint-staged` passes the list of files.
+
+It is important to note that `lint-staged` matches files to `glob` patterns in parallel and runs tasks on those matched groups in parallel as well. Within one matched group, the commands are executed in-order. There is an option to force `lint-staged` to process groups sequentially if you need to. In the above example `.scss` files will be processed in parallel with `.js` and `.ts` files. For `.js` and `.ts` files `eslint` will be executed first and only after it is done will `prettier` be executed.
+
+Check the following chapters for specifics about Prettier, ESLint and Stylelint.
+
+### Automatic code formatting with Prettier and IDE settings
+
+![Bikeshedding](../img/work.png)
+
+_Source: [XKCD](https://xkcd.com/1741/)_
+
+Many people are very passionate about the way they format their code. While we appreciate everyone's opinion, we believe it is best to leave this bikeshedding to automated tooling. It might not format the code in a way that is satisfying to everyone but it will be consistent across projects and, more importantly, it will format the code written by different people in the same way. This also saves some time wasted on discussion about formatting.
+
+[Prettier](https://prettier.io/) is one of the most popular tools for this job. It is very opinionated and not very configurable. It might not be perfect, but it is a good way to ensure consistency when it comes to formatting and it format multiple different file types.
+
+Here is our recommended Prettier configuration:
+```js
+// .prettierrc.json
+{
+  "$schema": "http://json.schemastore.org/prettierrc",
+  "printWidth": 120,
+  "endOfLine": "lf",
+  "useTabs": true,
+  "arrowParens": "always",
+  "quoteProps": "as-needed",
+  "bracketSpacing": true,
+  "singleQuote": true,
+  "semi": true,
+  "trailingComma": "es5",
+}
+```
+
+If you are using Angular, make sure to add an override for templates parser:
+
+```js
+// addendum to the above .prettierrc.json
+{
+  ...
+  "overrides": [{
+    "files": "*.component.html",
+    "options": {
+      "parser": "angular"
+    }
+  }]
+}
+```
+
+If you just added Prettier to an existing codebase, you should probably run it once and let it format the whole codebase. This will probably create a lot of modifications which you can skim through and if it checks out you can commit the changes. It is possible that some things might break after Prettier is run. In particular, we noticed some issues with the way Prettier formats SCSS, so you might want to exclude SCSS from Prettier:
+
+```bash
```

Comment:
I also had to tell VSCode not to format SCSS files when saving. I have the following the .vscode/settings.json:
```json
{
  "editor.formatOnSave": true,
  "[css]": {
    "editor.formatOnSave": false
  },
  "[scss]": {
    "editor.formatOnSave": false

  }
}
```

Reply:
Did you have .prettierignore file with scss ignored?